types/test.tsx

```diff
@@ -0,0 +1,317 @@
+/* tslint-disable */
+// TypeScript Version: 2.7
+
+import * as React from 'react';
+import {
+  Active,
+  Input,
+  Counter,
+  Focus,
+  FocusManager,
+  Form,
+  Hover,
+  List,
+  Map,
+  Set,
+  State,
+  Toggle,
+  Touch,
+  Value,
+  ListRender,
+  SetRender,
+  MapRender,
+  StateRender,
+  ToggleRender,
+  TouchRender,
+  ValueRender,
+  ActiveRender,
+  InputRender, CounterRender, FocusRender, FocusManagerRender, FormRender, HoverRender,
+} from './index';
+
+const noop = () => null;
+
+/* Active */
+{
+  const render: ActiveRender = ({active, bind}) => {
+    return null;
+  };
+  const onChange = (active: boolean) => {
+  };
+  [
+    <Active render={render}/>,
+    <Active>{render}</Active>,
+    <Active onChange={onChange} render={noop}/>,
+    <Active onChange={onChange}>{noop}</Active>,
+    // $ExpectError
+    <Active/>,
+  ];
+}
+
+/* Input */
+{
+  const render: InputRender = ({value, set, bind}) => {
+    return null;
+  };
+  const onChange = (value: string) => {
+    }
+  ;[
+  <Input render={render}/>,
+  <Input>{render}</Input>,
+  <Input onChange={onChange} render={noop}/>,
+  <Input onChange={onChange}>{noop}</Input>,
+  <Input initial={''} render={noop}/>,
+  // $ExpectError
```

Comment:
Does this work for typescript too?

Reply:
Yep, sure. It's a bit different from Flow, but almost the same.